### ISSUE TITLE
Read The Docs failing when used with tensorflow>=1.5

### ISSUE THREAD

**Issue by rtavenar** [rtavenar/tslearn#30]
I have not been able to build the docs under RTD with tensorflow version higher than 1.4.

I get the following error message:

```
python /home/docs/checkouts/readthedocs.org/user_builds/tslearn/envs/latest/bin/sphinx-build -T -E -b readthedocs -d _build/doctrees-readthedocs -D language=en . _build/html
Running Sphinx v1.7.4
loading translations [en]... done
making output directory...
[autosummary] generating autosummary for: auto_examples/index.rst, auto_examples/plot_barycenter_interpolate.rst, auto_examples/plot_barycenters.rst, auto_examples/plot_dtw.rst, auto_examples/plot_kernel_kmeans.rst, auto_examples/plot_kmeans.rst, auto_examples/plot_kshape.rst, auto_examples/plot_lb_keogh.rst, auto_examples/plot_neighbors.rst, auto_examples/plot_sax.rst, ..., gen_modules/tslearn.utils.rst, gen_modules/utils/tslearn.utils.load_timeseries_txt.rst, gen_modules/utils/tslearn.utils.save_timeseries_txt.rst, gen_modules/utils/tslearn.utils.to_time_series.rst, gen_modules/utils/tslearn.utils.to_time_series_dataset.rst, gen_modules/utils/tslearn.utils.ts_size.rst, gettingstarted.rst, index.rst, reference.rst, variablelength.rst
Illegal instruction (core dumped)
```

It seems related to the CPUs at RTD not having AVX functionalities, but I guess there should be a way to make things work.

The thing is:

1. we could downgrade TF to 1.4 for RTD, but it seems if we do not put a constraint on keras version, then there will be incompatibilities (eg. keras `sigmoid` calling `tf.nn.sigmoid` with a keyword argument `axis` that did not exist in TF 1.4): hence we should check which keras version could be OK
2. ideally, we would like to have a way to run latest TF from PyPI on RTD

Definitely, even if we choose option 1. for short-term fix, we should investigate option 2. because we might need new `keras` functionalities in the future.

**Comment by rtavenar** [comment]
I just implemented option 1 (TF1.4, keras2.0.8) as a temporary work-around. Docs are back, but I'm still looking for a way to implement option 2.

**Comment by ixjlyons** [comment]
If the docs build on Travis without any workarounds, another option might be to use [Doctr](https://drdoctr.github.io/doctr/) to deploy to GitHub Pages instead.

**Comment by rtavenar** [comment]
@LevavasseurYann sorry, I don't get what's not working for you. Could you explain it in more details?
